airflow/jobs.py

```diff
@@ -721,6 +721,25 @@ def create_dag_run(self, dag, session=None):
             if dag.schedule_interval == '@once' and last_scheduled_run:
                 return None
 
+            # don't do backfill for dag's that don't have dag.backfill = True
+            if not dag.backfill:
+                # The logic is that we move start_date up until
+                # one period before, so that datetime.now() is AFTER
+                # the period end, and the job can be created...
```

Comment:
@btallman I think I agree with @zodiac, just because we store the start_date in the DB and it is kind of assumed then it is immutable. On the other hand it does add to the complexity.

What you could do is use a local start_date that is set to dag.start_date at the beginning of the function and use that down the line?